pyls/language_server.py

```diff
@@ -75,22 +77,28 @@ class LanguageServer(MethodJSONRPCServer):
     """
 
     process_id = None
-    root_path = None
+    root_uri = None
     init_opts = None
 
     def capabilities(self):
         return {}
 
-    def initialize(self, root_path, init_opts, process_id):
+    def initialize(self, root_uri, init_opts, process_id):
         pass
 
     def m_initialize(self, **kwargs):
         log.debug("Language server intialized with %s", kwargs)
-        self.root_path = kwargs.get('rootPath')
+        if 'rootUri' in kwargs:
+            self.root_uri = kwargs.get('rootUri')
+        elif 'rootPath' in kwargs:
+            root_path = kwargs.get('rootPath')
+            self.root_uri = urljoin(u'file:', pathname2url(root_path))
+        else:
+            self.root_uri = ''
```

Comment:
Does this need to be a unicode literal? I'm not actually sure... Equally, what do we get from the kwargs?